src/train_model.py

```diff
@@ -12,37 +12,47 @@
 from sacred.observers import FileStorageObserver
 from sacred.utils import apply_backspaces_and_linefeeds
 
-from add_mul import AddMul
-from clusterability_gradient import LaplacianEigenvalues
-from networks import cnn_dict, mlp_dict
-from tiny_dataset import TinyDataset
-from utils import (
+from src.add_mul import AddMul
+from src.clusterability_gradient import LaplacianEigenvalues
+from src.networks import cnn_dict, mlp_dict
+from src.simple_data_loader import SimpleDataLoader
+from src.tiny_dataset import TinyDataset
+from src.utils import (
     get_weight_modules_from_live_net,
     size_and_multiply_np,
     size_sqrt_divide_np,
     vector_stretch,
 )
 
-train_exp = Experiment('train_model')
+train_exp = Experiment('train_model', interactive=True)
 train_exp.captured_out_filter = apply_backspaces_and_linefeeds
 train_exp.observers.append(FileStorageObserver('training_runs'))
 
 # probably should define a global variable for the list of datasets.
 # maybe in a utils file or something.
 
-# TODO: have a function that tells you the dimensions of the stuff in your
-# dataset
+SIMPLE_FUNCTIONS = {
+    "high_freq_waves":
+    [lambda x: torch.sin(5 * x), lambda x: torch.cos(5 * x)],
+    "many_fns":
+    [torch.sin, lambda x: torch.log(x + 5 + 1e-3), torch.cos, torch.exp],
+    "many_fns_norm": [
+        lambda x: torch.sin(x) / 0.72605824,
+        lambda x: torch.log(x + 5 + 1e-3) / 0.996789,
+        lambda x: torch.cos(x) / 0.6598921, lambda x: torch.exp(x) / 29.747263
+    ],
+}
 
 
 @train_exp.config
 def mlp_config():
     batch_size = 128
     num_epochs = 20
     log_interval = 100
-    dataset = 'kmnist'
+    dataset = 'simple_dataset'
     model_dir = './models/'
     net_type = 'mlp'
-    net_choice = 'small'
+    net_choice = 'simple'
```

Comment:
seems like this + the dataset variable should be reverted, and you could put your versions in a special named_config.